generator/webapp/src/main/java/io/fabric8/maven/generator/webapp/WebAppGenerator.java

```diff
@@ -68,7 +73,30 @@ public WebAppGenerator(GeneratorContext context) {
     @Override
     public boolean isApplicable(List<ImageConfiguration> configs) {
         return shouldAddImageConfiguration(configs) &&
-               MavenUtil.hasPlugin(getProject(), "org.apache.maven.plugins:maven-war-plugin");
+               MavenUtil.hasPlugin(getProject(), "org.apache.maven.plugins:maven-war-plugin") && !isWarUsedWithFatJar(configs);
+    }
+
+    /**
+     * If we are using spring boot or wildfly swarm with a fat jar then we should not use the web app generator
+     * and should stick with the fat jar generators instead
+     */
+    protected boolean isWarUsedWithFatJar(List<ImageConfiguration> configs) {
+        MavenProject project = getContext().getProject();
+        String finalName = project.getBuild().getFinalName();
+        String artifactId = project.getArtifactId();
+        if (!Objects.equal(artifactId, finalName)) {
+            String plugin = null;
+            if (MavenUtil.hasPlugin(project, SPRING_BOOT_MAVEN_PLUGIN_GA)) {
```

Comment:
That's what I thought, too (except it the option [add](https://github.com/rhuss/fabric8-maven-plugin/blob/20cb6e08917a644b5c128d1d5489624cd71e5cbf/generator/api/src/main/java/io/fabric8/maven/generator/api/support/BaseGenerator.java#L61-L61) is set to `true` in which case generator generator configs as accumulated. otherwise only the first wins). If you give me a bit of time until, I'm happy to check this as the first thing in the morning tomorrow.